### PR TITLE
WW-5445 Adds support for Struts 6

### PR DESCRIPTION
**Pull request by lukaszlenart**
Closes [WW-5445](https://issues.apache.org/jira/browse/WW-5445)

### CODE REVIEW

**Comment by lukaszlenart** [comment]
@burtonrhodes could you review and approve?

**Comment by lukaszlenart** [comment]
A new test version is here https://dist.apache.org/repos/dist/dev/struts/idea-plugin/

**Comment by burtonrhodes** [comment]
The filesets are now loading.  Looks good. Thanks!

**Comment by burtonrhodes** [comment]
The only thing I see that might be worth adding for consistency is registering a DTD resource in Struts2ResourceProvider.java

`
addDTDResource(StrutsConstants.STRUTS_6_0_DTD_URI, 
   StrutsConstants.STRUTS_6_0_DTD_ID,
   "struts-6.0.dtd", registrar);
`

You would then need to add a `struts-6.0.dtd` file in the `resources.dtds` directory

**Comment by lukaszlenart** [comment]
> Lastly, it appears none of the jsp file entries are clickable (red) in my custom struts.xml file. I'm not sure why this is not working for me (could be a intellij setting?), but perhaps I'll file a separate Jira ticket.

I see the same problem yet some of the JSP files are clickable

**Comment by lukaszlenart** [comment]
What's the version IDEA you use?

**Comment by burtonrhodes** [comment]
The latest version:

IntelliJ IDEA 2024.1.4 (Ultimate Edition)
Build #IU-241.18034.62

**Comment by lukaszlenart** [comment]
This can be the case, if I upgrade plugin dependency to this version it won't build, there is a bit of deprecated/missing things

File: build.gradle.kts [hunk 1]
```diff
@@ -44,7 +44,7 @@ tasks {
 
     patchPluginXml {
         sinceBuild.set("231")
-        untilBuild.set("233.*")
+        untilBuild.set("241.*")
```
Comment (lukaszlenart) [inline build.gradle.kts:47]: I will revert to the latest 23x. build

**Comment by burtonrhodes** [comment]
Thanks @lukaszlenart   Latest build seems to load fine and I am able to select Filesets.  However, the my jsp links in my struts.xml config files are still showing up red.  In other words, the associated JSP file for the <result> is not matched with a file in my directory structure (and is not clickable).  Wondering if this is a configuration issue on my end?

Screenshot 2024-08-14 095414

**Comment by lukaszlenart** [comment]
Yes, I noticed that, but looks like all of them are navigable. I'm going to merge this PR and address the JSP issue in another one.

**Pull request merged**